File: src/AutoRest.SdkExplorer/Model/Schema/SchemaEnumValue.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace AutoRest.SdkExplorer.Model.Schema
{
    public class SchemaEnumValue
    {
        public string? Value { get; set; }
        public string? InternalValue { get; set; }
        public string Description { get; set; } = string.Empty;

        public SchemaEnumValue()
        {

        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Schema/SchemaEnumValue.ts
import { SchemaBase } from "./SchemaBase";

export class SchemaEnumValue implements AutoRest.SdkExplorer.Interface.SchemaEnumValue {
    value: string;
    internalValue?: string;
    description: string;

    constructor(data: AutoRest.SdkExplorer.Interface.SchemaEnumValue) {
        this.value = data.value?.toString()!;
        this.internalValue = data.internalValue?.toString();
        this.description = data.description ?? "";
    }
}
